File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
def solution(s):
    answer = 0
    
    stack = []
    
    for data in s:
        if stack:
            if stack[-1] == data:
                stack.pop()
            else:
                stack.append(data)
        else:
            stack.append(data)
        
    
    if not stack:
        answer = 1
    
    return answer

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.Stack;

class Solution
{
    public int solution(String s)
    {
        int answer = 0;
		
		Stack<Character> stack = new Stack<Character>();
		
		for(int i=0; i<s.length(); i++) {
			if(!stack.isEmpty() && stack.peek() == s.charAt(i)) {
				stack.pop();
				continue;
			}
			stack.add(s.charAt(i));
		}
		
		
		answer = stack.isEmpty() == true ? 1 : 0;

        return answer;
    }
}